src/monattack.cpp

```diff
@@ -614,7 +614,9 @@ bool mattack::shockstorm( monster *z )
         auto msg_type = target == &g->u ? m_bad : m_neutral;
         add_msg( msg_type, _( "A bolt of electricity arcs towards %s!" ), target->disp_name().c_str() );
     }
-    sfx::play_variant_sound( "fire_gun", "bio_lightning", sfx::get_heard_volume( z->pos() ) );
+    if( !g->u.is_deaf() ) {
+        sfx::play_variant_sound( "fire_gun", "bio_lightning", sfx::get_heard_volume( z->pos() ) );
```

Comment:
This doesn't seem to be the right way to fix that. I'd replace the raw `sfx` function with a more high-level one which takes deafness (along with many other things) into account:

```c++
sounds::sound( pos(), ?, sounds::sound_t::combat, _( "?" ), false, "fire_gun", "bio_lightning" );
```
Question marks denote the fields I'm not sure about.

Reply:
Maybe we should just put deafness check to `get_heard_volume`?

Reply:
This wouldn't be enough. `sounds::sound` is a bit messy (well, not a bit), but it does some important things apart from checking for deafness. It can:
- Deafen you if the sound is deafening and you're not immune;
- Wake you up (with respect to sleeping-related perks and effects);
- Describe the sound for you depending on whether you can see the source of it or not;
- Output a prompt if you're performing an activity;
- Store a sound marker if the source was unseen (footprints);
- Play the actual sound. BTW without checking for being deaf, and it also lacks pitch. So answering your question: we probably should as well.

Reply:
I'm going to merge this is a temporary solution and replace occurrences of `sfx::play_variant_sound` with `sounds::sound` where appropriate in a follow-up PR.